src/box/alter.cc

```diff
@@ -4078,10 +4079,42 @@ on_commit_replicaset_name(struct trigger *trigger, void * /* event */)
 	return 0;
 }
 
+static int
+on_commit_dd_version_start_filtering(va_list /* ap */)
+{
+	txn_limbo_filter_enable(&txn_limbo);
+	return 0;
+}
+
+static int
+on_commit_dd_version_stop_filtering(va_list /* ap */)
+{
+	txn_limbo_filter_disable(&txn_limbo);
+	return 0;
+}
+
 static int
 on_commit_dd_version(struct trigger *trigger, void * /* event */)
```

Comment:
Yes, added one:
```diff
diff --git a/src/box/alter.cc b/src/box/alter.cc
index d09b59fcda..f240a8acd1 100644
--- a/src/box/alter.cc
+++ b/src/box/alter.cc
@@ -4093,6 +4093,10 @@ on_commit_dd_version_stop_filtering(va_list /* ap */)
        return 0;
 }
 
+/**
+ * Update the cached schema version and enable version-dependent features, like
+ * split-brain detection.
+ */
 static int
 on_commit_dd_version(struct trigger *trigger, void * /* event */)
 {

```

By the way, I'm thinking maybe we shouldn't bother with enabling split-brain detection right away on upgrade? Maybe let it be enabled only on restart? This way we won't need these ugly triggers on schema version change.

Reply:
Can't it fail before restarting? I think we should enable it as soon as possible, because as far as I remember the whole reason for split-brain detection is to alert the user as soon as possible, that split brain occurred.

Reply:
Yes, it sure can. But the split-brain could happen and go undetected before 2.10 as well. So IMO it's not that important to enable the detection ASAP on upgrade.

Although, saying that the feature starts working not after the actual upgrade, and not after schema upgrade, but after all that and a restart, might be too complicated for the user. So now I lean towards keeping the triggers.

Reply:
Do we require a restart after schema upgrading for anything else to start working? If not - I think we should leave the triggers here. If we do need a restart anyway - than no reason in keeping them, and the upgrade procedure must be something like:

1. Stop old version
2. Start new version
3. Upgrade schema
4. Restart new version

Reply:
No, there's no need to restart the nodes after upgrading schema to enable other features, AFAIR.
Let's keep the triggers then.